src/Tools/CodeGenerator/Featurizers.yaml

```diff
@@ -1392,55 +1396,59 @@ featurizers:
 
       C++-style pseudo signature:
-        template <typename T> std::vector<T> execute(std::tuple<std::vector<std::string> const &, T const &> value);
+        template <typename T> matrix<T> execute(std::tuple<std::vector<std::string> const &, T const &> value);
 
       Examples:
           A simple example would be horizon = 1, maxWindowSize = 2, and we want to take the minimum.
 
           +-----------+-------+-------------------+
           | grain     | target| target_minimum    |
           +===========+=======+===================+
-          | A         | 10    | [NAN]             |
+          | A         | 10    | [[NAN]]           |
           +-----------+-------+-------------------+
-          | A         | 4     | [10]              |
+          | A         | 4     | [[10]]            |
           +-----------+-------+-------------------+
-          | A         | 6     | [4]               |
+          | A         | 6     | [[4]]             |
           +-----------+-------+-------------------+
-          | A         | 11    | [4]               |
+          | A         | 11    | [[4]]             |
           +-----------+-------+-------------------+
 
           A more complex example would be, assuming we have horizon = 2, maxWindowSize = 2, minWindowSize = 2, and we want the maximum value
           +-----------+-------+-------------------+
           | grain     | target| target_max        |
           +===========+=======+===================+
-          | A         | 10    | [NAN, NAN]        |
+          | A         | 10    | [[NAN, NAN]]      |
           +-----------+-------+-------------------+
-          | A         | 4     | [NAN, NAN]        |
+          | A         | 4     | [[NAN, NAN]]      |
           +-----------+-------+-------------------+
-          | A         | 6     | [NAN, 10]         |
+          | A         | 6     | [[NAN, 10]]       |
           +-----------+-------+-------------------+
-          | A         | 11    | [10, 6]           |
+          | A         | 11    | [[10, 6]]         |
           +-----------+-------+-------------------+
 
     templates:
       - name: T
         types:
-          - int8
-          - int16
-          - int32
-          - int64
-          - uint8
-          - uint16
-          - uint32
-          - uint64
+          # Note that optional values during output increase quite a bit of overhead when marshalling, therefore
+          # this featurizer only supports those types that don't have to pay that penality. We may still want to support
+          # these types in the future in some frameworks.
+          #
+          # - int8
+          # - int16
+          # - int32
+          # - int64
+          # - uint8
+          # - uint16
+          # - uint32
+          # - uint64
+          # - string
           - float
           - double
-          - string

```

Comment:
If are making the change to only do float and double, just like the analytical rolling window, does it make sense to leave these as 2 separate featurizers? @davidbrownellWork @angryjinyan

Reply:
It does make sense to leave them as 2 different featurizers, as the limitation is due to marshaling. A hypothetical C++ framework could directly support these optional types.

Reply:
Makes sense to me. I just synced with Jin ofline on this at the same time you responded as well. I hadn't thought about it from that perspective.